### PR TITLE
Message consolidation

### PR DESCRIPTION
**Pull request by amotaouakkil123**
Created the power tab that sends a powerCmd message with the appropriate "power on" messages. Power tab is updated with all the correct messages and the main UI .py file contains all the correct feedback messages alongside the right publishers. 
Need to implement the servo publishers.

### CODE REVIEW

File: ui_app/src/power_backend.py [hunk 1]
```diff
@@ -1,34 +1,93 @@
...
+        self.science_on = 0.0
+        self.lower_arm_on = 0.0
+        self.upper_arm_on = 0.0
+
+        self.system_publisher = Float32MultiArray()
+        self.system_publisher.data.append(self.drive_on)
+        self.system_publisher.data.append(self.science_on)
+        self.system_publisher.data.append(self.lower_arm_on)
+        self.system_publisher.data.append(self.upper_arm_on)
+
```
Comment (Shaswata79) [inline ui_app/src/power_backend.py:21]: Include the servo angles here?

File: drive_control/src/drive_control_node.py [hunk 2]
```diff
@@ -42,6 +44,8 @@ def __init__(self):
         rospy.init_node('drive_controller')
         self.angular_velocity_publisher = rospy.Publisher('/wheel_velocity_cmd', WheelSpeed, queue_size=1)
         self.robot_twist_subscriber = rospy.Subscriber("rover_velocity_controller/cmd_vel", Twist, self.twist_to_velocity)
+        self.motor_publisher = rospy.Subscriber("/driveCmd", Float32MultiArray, queue_size=1)
```
[the remark this answers is not in the document]
Comment (amotaouakkil123) [reply drive_control/src/drive_control_node.py:47]: It's supposed to be a publisher, I'm publishing straight from the drive_control_node

File: drive_control/src/drive_control_node.py [hunk 3]
```diff
@@ -42,6 +44,8 @@ def __init__(self):
         rospy.init_node('drive_controller')
         self.angular_velocity_publisher = rospy.Publisher('/wheel_velocity_cmd', WheelSpeed, queue_size=1)
```
[the remark this answers is not in the document]
Comment (amotaouakkil123) [reply drive_control/src/drive_control_node.py:45]: No I still want to have access to the wheel_velocities before publishing the motor values.

**Review by acoulombe — CHANGES_REQUESTED** [review]

File: ui_app/src/power_backend.py [hunk 4]
```diff
@@ -1,34 +1,93 @@
...
+        self.power_on_publisher = rospy.Publisher("/powerCmd", Float32MultiArray, queue_size=1)
 
     @staticmethod
     def update_float_value(value, label):
         label.display("%.2f" % float(value))
+    
+    def get_drive_enabled(self):
+        if self.ui.drive_enabled.isChecked():
```
[the remark this answers is not in the document]
Comment (amotaouakkil123) [reply ui_app/src/power_backend.py:29]: Is this necessary? I think the function naming is clear enough and also works when testing the ticking of the boxes.

[the remark this answers is not in the document]
Comment (acoulombe) [reply ui_app/src/power_backend.py:29]: It's a suggestion for cleaner code. You can convert the if condition to a simple read

File: drive_control/src/drive_control_node.py [hunk 5]
```diff
@@ -99,9 +104,16 @@ def run(self):
...
+            motor_val.data.append(motor_lf)
+            motor_val.data.append(motor_rb)
+            motor_val.data.append(motor_rf)
+
+
             
             print(f"Motor values: {[motor_lf, motor_lb, motor_rf, motor_rb]}")
 
+
             self.angular_velocity_publisher.publish(cmd)
```
[the remark this answers is not in the document]
Comment (amotaouakkil123) [reply drive_control/src/drive_control_node.py:114]: Btw any changes on drive_control is on Joseph's branch for drive and gamepad process changes.

**Pull request merged**